test/Transactions/Orleans.Transactions.Azure.Test/AzureTransactionLogTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage.Table;
+using Orleans.Configuration;
+using Orleans.Runtime.Configuration;
+using Orleans.Transactions.Abstractions;
+using TestExtensions;
+using Xunit;
+
+namespace Orleans.Transactions.AzureStorage.Tests
+{
+    [TestCategory("Azure"), TestCategory("Transactions"), TestCategory("BVT")]
+    public class AzureTransactionLogTests 
+    {
+        private const string TestClusterId = "AzureTransactionLogTestCluster";
+        private static string TableName = $"TransactionLog{((uint) Guid.NewGuid().GetHashCode()) % 100000}";
+        public AzureTransactionLogTests()
+        {
+            TestFixture.CheckForAzureStorage(TestDefaultConfiguration.DataConnectionString);
+        }
+
+        [SkippableFact]
+        public async Task TransactionLogCanArchiveAndQuery()
+        {
+            var azureOptions = new AzureTransactionLogOptions()
+            {
+                ArchiveLog = true,
+                ConnectionString = TestDefaultConfiguration.DataConnectionString,
+                TableName = TableName
+            };
+
+            var logStorage = await StorageFactory(azureOptions);
+            var recordsNum = 2000;
+            var allTransactions = new List<CommitRecord>(recordsNum);
+            for (int i = 0; i< recordsNum; i++)
+            {
+                allTransactions.Add(new CommitRecord(){LSN=i, TransactionId = i});
+            }
+            await logStorage.Append(allTransactions);
+            //a subset of all transactions will be archived 
+            await logStorage.TruncateLog(recordsNum + 2);
+            var query = new TableQuery<AzureTransactionLogStorage.ArchivalRow>().Where(
+                TableQuery.GenerateFilterCondition(nameof(AzureTransactionLogStorage.ArchivalRow.ClusterId), QueryComparisons.Equal, TestClusterId));
+            var archivalTransactions = await logStorage.QueryArchivalRecords(query);
+            //assert a subset of transaction has been archived and no duplicates
+            Assert.True(allTransactions.Select(tx=>tx.LSN).ToImmutableHashSet().IsSubsetOf(archivalTransactions.Select(tx=>tx.LSN)));
+            Assert.Equal(archivalTransactions.Select(tx => tx.LSN).Count(), archivalTransactions.Select(tx => tx.LSN).ToImmutableHashSet().Count);
```

Comment:
unclear what the expected outcome is.  Asserts usually have (expected, actual) with expected state being a clear an comprehensible state.  In this case it looks like we're comparing the actual count with the actual count..?

Reply:
this line probably not needed. but we are comparing no duplicared LSN, which I commented above

Reply:
archivalTransactions.Select(tx => tx.LSN).Count() returns number of archivalTransactions, correct?
archivalTransactions.Select(tx => tx.LSN).ToImmutableHashSet().Count()  returns number of archivalTransactions, correct?
So we're comparing the same things..?

Reply:
if a array is {1,1,2}, ToHashSet would make it {1, 2}. So their count would be different. 

it shouldn't happen with this case since LSN isn't the row key. and there won't be duplicated two row with the same row key. I can remove this line, but just they are not comparing the same thing. Or I missed what ToHashSet do ?